viewer/src/main/java/nl/b3p/viewer/util/IPAuthenticationFilter.java

```diff
@@ -0,0 +1,347 @@
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletRequestWrapper;
+import javax.servlet.http.HttpSession;
+import nl.b3p.viewer.config.security.User;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.hibernate.Hibernate;
+import org.stripesstuff.stripersist.Stripersist;
+
+/**
+ *
+ * @author Meine Toonen meinetoonen@b3partners.nl
+ */
+public class IPAuthenticationFilter implements Filter {
+    
+    private static final Log log = LogFactory.getLog(IPAuthenticationFilter.class);
+
+    // The filter configuration object we are associated with.  If
+    // this value is null, this filter instance is not currently
+    // configured. 
+    private FilterConfig filterConfig = null;
+    
+    private static final String IP_CHECK = IPAuthenticationFilter.class + "_IP_CHECK";
+    private static final String USER_CHECK = IPAuthenticationFilter.class + "_USER_CHECK";
+    
+    
+    public IPAuthenticationFilter() {
+    }    
+    
+    /**
+     *
+     * @param r The servlet request we are processing
+     * @param response The servlet response we are creating
+     * @param chain The filter chain we are processing
+     *
+     * @exception IOException if an input/output error occurs
+     * @exception ServletException if a servlet error occurs
+     */
+    public void doFilter(ServletRequest r, ServletResponse response,
+            FilterChain chain)
+            throws IOException, ServletException {
+        
+        log.debug("IPAuthenticationFilter:doFilter()");
+        HttpServletRequest request = (HttpServletRequest) r;
+        HttpSession session = request.getSession();
+        User u = null;
+        if(session.getAttribute(IP_CHECK) == null  && session.getAttribute(USER_CHECK) == null){
+            
+            String ipAddress = request.getHeader("X-FORWARDED-FOR");
+            if (ipAddress == null) {
+                ipAddress = request.getRemoteAddr();
+            }
+            session.setAttribute(IP_CHECK, ipAddress);
+            Stripersist.requestInit();
+            
+            EntityManager em = Stripersist.getEntityManager();
+            List<User> users = em.createQuery("from User", User.class).getResultList();
+            List<User> possibleUsers = new ArrayList<User>();
+            
+            for (User user : users) {
+                if(checkValidIpAddress(request, user)){
+                    possibleUsers.add(user);
+                }
+            }
+            
+            if(possibleUsers.isEmpty()){
+                log.debug("No possible users found for ip");
+            }else if( possibleUsers.size() == 1){
+                u = possibleUsers.get(0);
+                Hibernate.initialize(u.getGroups());
+                session.setAttribute(IP_CHECK, ipAddress);
+                session.setAttribute(USER_CHECK, u);
+            }else{
+                log.debug("Too much possible users found for ip.");
```

Comment:
Too many... also I would show this as a warning or error as it is a configuration problem